lib/plugins/aws/package/compile/events/sns/index.js

```diff
@@ -225,10 +225,30 @@ class AwsCompileSNSEvents {
             });
 
             if (event.sns.redrivePolicy && event.sns.redrivePolicy.deadLetterTargetArn) {
-              const queuePolicyLogicalId = this.provider.naming.getQueueLogicalId(
+              const queuePolicyLogicalId = this.provider.naming.getTopicDLQPolicyLogicalId(
                 functionName,
-                `${topicName}DLQ`
+                topicName
               );
+
+              let queueUrl;
+              const arnDefinition = event.sns.redrivePolicy.deadLetterTargetArn;
+              if (typeof arnDefinition === 'object' && arnDefinition['Fn::GetAtt']) {
+                queueUrl = { Ref: arnDefinition['Fn::GetAtt'][0] };
+              } else if (typeof arnDefinition === 'string' && arnDefinition.startsWith('arn:')) {
+                // obtain queue url from an arn like arn:aws:sqs:us-east-2:444455556666:queue1
+                const queueParts = arnDefinition.split(':');
+                const partsLength = queueParts.length;
+                queueUrl = `https://sqs.${queueParts[partsLength - 3]}.amazonaws.com/${
```

Comment:
Rather than `amazonaws.com`, can you take advantage of [`AWS::URLSuffix`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html#cfn-pseudo-param-urlsuffix)?

Something like
```javascript
queueUrl = {
  'Fn::Join': [
    '',
    `https://sqs.${queueParts[partsLength - 3]}.`,
    { Ref: 'AWS::URLSuffix' },
    `/${queueParts[partsLength - 2]}/${queueParts[partsLength - 1]}`
  ]
};

```

Reply:
done!